backend/src/database/helpful_links.go

```diff
@@ -0,0 +1,95 @@
+package database
+
+import (
+	"UnlockEdv2/src/models"
+	"fmt"
+)
+
+func (db *DB) GetHelpfulLinks(page, perPage int, search, orderBy string) (int64, []models.HelpfulLink, error) {
+	links := make([]models.HelpfulLink, 0, perPage)
+	tx := db.Model(&models.HelpfulLink{})
+	var total int64
+
+	validOrder := map[string]bool{
+		"title ASC":       true,
+		"title DESC":      true,
+		"created_at ASC":  true,
+		"created_at DESC": true,
+	}
+
+	if search != "" {
+		search = "%" + search + "%"
+		tx = tx.Where("LOWER(title) LIKE ?", search)
+	}
+
+	if valid, ok := validOrder[orderBy]; ok && valid {
+		tx = tx.Order(orderBy)
+	} else {
+		tx = tx.Order("created_at DESC")
+	}
+
+	if err := tx.Count(&total).Error; err != nil {
+		return 0, nil, newGetRecordsDBError(err, "helpful_links")
+	}
+
+	if err := tx.Offset(calcOffset(page, perPage)).Limit(perPage).Find(&links).Error; err != nil {
+		return 0, nil, newGetRecordsDBError(err, "helpful_links")
+	}
+
+	return total, links, nil
+}
+func (db *DB) AddHelpfulLink(link models.HelpfulLink) error {
```

Comment:
can we pass the struct by reference to add + edit? (just dont forget to remove the `&` after e.g. `Create(&link)`)